src/app/actions/course.ts

```diff
@@ -47,13 +47,49 @@ export async function coursesBy(
     return {};
   };
 
-  const courses = await prisma.course.findMany({
-    where: {
-      departmentCode: departmentCodeParam().departmentCode,
-      number: numberParam().number,
-      ge: geParam().ge,
-    },
-  });
+  const creditParam = () => {
+    if (pred.creditRange) {
+      return {
+        credits: {
+          gte: pred.creditRange[0],
+          lte: pred.creditRange[1],
+        },
+      };
+    }
+    return {};
+  };
+
+  let courses = [];
```

Comment:
A 1-2 line comment might be helpful here to help other devs with understanding what this code does